### PR TITLE
[www] feat(starters-showcase): Improve performance

### PR DESCRIPTION
**Pull request by EmaSuriano**
Hi, Gatsby community! 👋 

I love the new design of the showcase for Gatsby starter but one thing I noticed was the bad performance when trying to filter using the search input on the right. The problem was that every time the user type on the input, all the Gatsby starter is filtered.

The solution I made is to add some `debounce` magic on every `onChange` and also move the value of the input to the `state`. Now on every keystroke, the application re-render only the `input` and after 500 ms has passed after the `onChange` call I filter the starter list.

That's all folks, hope it helps! 😄

### CODE REVIEW

**Comment by fk** [comment]
💜🙏

**Comment by EmaSuriano** [comment]
@amberleyromo yeah sure, I didn't notice the component folder. Also, I will add some tests so it doesn't decrease coverage 😅

**Comment by amberleyromo** [comment]
I can hear @DSchau smiling already :)

**Comment by EmaSuriano** [comment]
I moved `debounce-input` to components folder. Unfortunately, there was any `test` command in the `package.json`, so I didn't write tests because nobody is going to run it 😂 If there is a plan to add any testing framework, I'm in to write a few cases for `debounce-input`.

File: www/src/views/starter-library/filtered-starters.js [hunk 1]
```diff
@@ -39,6 +40,11 @@ export default class FilteredStarterLibrary extends Component {
   resetFilters = () =>
     this.props.setURLState({ c: null, d: null, v: null, s: `` })
 
+  onChangeUrlWithText = e => {
+    console.log(e)
```
Comment (DSchau) [inline www/src/views/starter-library/filtered-starters.js]: Let's remove this!

Comment (EmaSuriano) [reply www/src/views/starter-library/filtered-starters.js]: Ooops, sorry! Removed 😉

**Comment by DSchau** [comment]
@EmaSuriano thanks so much for your commit. We're happy to have you contributing :)

**Pull request merged**